### PR TITLE
Move from jbuilder to dune.

### PR DESCRIPTION
**Pull request by kc284**
Signed-off-by: Konstantina Chremmou <konstantina.chremmou@citrix.com>

### CODE REVIEW

File: lib_gen/types/dune [hunk 1]
```diff
@@ -0,0 +1,43 @@
+; pkg-config, can't use Configurator due to old OCaml version
+(rule (with-stdout-to "dlm_cflags" (run pkg-config libdlm --cflags)))
+(rule (with-stdout-to "dlm_libs" (run pkg-config libdlm --libs)))
```
Comment (edwintorok) [inline lib_gen/types/dune:3]: I  checked the dune docs, and they now recommend to use Configurator to generate these flags: https://github.com/ocaml/dune/blob/master/doc/configurator.rst.
I think our OCaml version is new enough now that we could use something like this (configurator is now part of dune itself):
```
(executable
 (name discover)
 (libraries dune.configurator))

(rule
 (targets dlm_libs.sexp dlm_cflags.sexp)
 (deps (:discover discover.exe))
 (action (run %{discover})))
```

```ocaml
module C = Configurator.V1

let () =
  C.main ~name:"ocaml-dlm" (fun c ->
      let conf = match C.Pkg_config.get c with
      | None -> failwith "pkg-config not installed"
      | Some pc ->
        match (C.Pkg_config.query pc ~package:"libdlm") with
        | None -> failwith "dlm C library is not installed"
        | Some deps -> deps in
      C.Flags.write_sexp "dlm_cflags.sexp" conf.cflags;
      C.Flags.write_sexp "dlm_libs.sexp" conf.libs);;
```

**Review by edwintorok — COMMENTED** [review]
This looks good, I think we can also avoid the `echo ()` workaround and use dune's builtin pkg-config support (albeit at the cost of slightly more boilerplate).

File: lib_gen/types/dune (hunk 1 above)
Comment (kc284) [reply lib_gen/types/dune:3]: We'll still need to write out the non-s-expression files `dlm_cflags` and `dlm_libs` though; they are used further down in the rule building types_stubgen.exe.

Comment (edwintorok) [reply lib_gen/types/dune:3]: Lets keep it as is for now then and improve it in the future if we find a solution that doesn't require the `dlm_libs` files.

**Review by edwintorok — APPROVED** [review]

**Pull request merged**